recipes/gdcm/all/conanfile.py

```diff
@@ -0,0 +1,228 @@
+import os
+import textwrap
+
+required_conan_version = ">=1.33.0"
+
+class GDCMConan(ConanFile):
+    name = "gdcm"
+    topics = ("dicom", "images")
+    homepage = "http://gdcm.sourceforge.net/"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "BSD-3-Clause"
+    description = "C++ library for DICOM medical files"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    generators = "cmake", "cmake_find_package"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def validate(self):
+        if (self.settings.compiler == "Visual Studio"
+                and self.options.shared
+                and str(self.settings.compiler.runtime).startswith("MT")):
+            raise ConanInvalidConfiguration("shared gdcm can't be built with MT or MTd")
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, "11")
+
+    def requirements(self):
+        self.requires("zlib/1.2.11")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["GDCM_BUILD_DOCBOOK_MANPAGES"] = "OFF"
+        self._cmake.definitions["GDCM_BUILD_SHARED_LIBS"] = "ON" if self.options.shared else "OFF"
+        self._cmake.definitions["GDCM_USE_SYSTEM_ZLIB"] = "ON"
+
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("Copyright.txt", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        if self.settings.os == "Windows":
+            bin_dir = os.path.join(self.package_folder, "bin")
+            tools.remove_files_by_mask(bin_dir, "[!gs]*.dll")
+            tools.remove_files_by_mask(bin_dir, "*.pdb")
+        lib_dir = os.path.join(self.package_folder, "lib")
+        tools.rmdir(os.path.join(lib_dir, "gdcmopenjpeg-2.3"))
+        tools.rmdir(os.path.join(lib_dir, "pkgconfig"))
```

Comment:
please add pkg_config targets!

Reply:
Sadly I have no idea how to work with pkgconfig. pkgconfig folder has only one file
https://github.com/inviwo/gdcm/blob/master/Utilities/gdcmopenjpeg/src/lib/openjp2/libopenjp2.pc.cmake.in which gets installed with absolute local path
Maybe it's better to try to reuse openjpeg 2.3.1 from CCI? It has
`self.cpp_info.names["pkg_config"] = "libopenjp2"`

Reply:
That looks pretty good to me

Reply:
I couldn't find any usages of gdcm through pkg_config